### PR TITLE
[MISC] Renovate tweaks

### PR DESCRIPTION
**Pull request by dragomirp**
* Add regex managers for juju agents
* pins libjuju 2 installed by CI
* Tweaks libjuju 3 pin in pyproject

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -33,9 +33,9 @@ jobs:
       fail-fast: false
       matrix:
         juju:
-          - agent: 2.9.46
-            libjuju: ^2
-          - agent: 3.1.7
+          - agent: 2.9.46  # renovate: latest juju 2
+            libjuju: ==2.9.46.1  # renovate: latest libjuju 2
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:37]: Currently we install the latest libjuju 2. Do we want to pin?

File: pyproject.toml [hunk 2]
```diff
@@ -44,7 +44,8 @@ pytest-github-secrets = {git = "https://github.com/canonical/data-platform-workf
 pytest-operator = "^0.34.0"
 pytest-operator-cache = {git = "https://github.com/canonical/data-platform-workflows", tag = "v13.0.0", subdirectory = "python/pytest_plugins/pytest_operator_cache"}
 pytest-operator-groups = {git = "https://github.com/canonical/data-platform-workflows", tag = "v13.0.0", subdirectory = "python/pytest_plugins/pytest_operator_groups"}
-juju = "^3.2.2"
+# renovate caret doesn't work: https://github.com/renovatebot/renovate/issues/26940
+juju = "<=3.4.0.0"
```
Comment (dragomirp) [inline pyproject.toml:48]: We either need to keep the fake semver, which will update the lock but won't change pyproject, or use less than.

File: .github/renovate.json5 [hunk 3]
```diff
@@ -1,12 +1,72 @@
...
     // Later rules override earlier rules
     {
       "matchPackageNames": ["python"],
       "allowedVersions": "<3.11"
+    },
+    {
+      "matchDepNames": ["Juju 2"],
+      "matchPackageNames": ["juju/juju"],
+      "allowedVersions": "<3.0.0",
+      "extractVersion": "^v(?<version>.*)$",
```
Comment (dragomirp) [inline .github/renovate.json5:15]: Release format changed from `juju-{version}` to `v{version}`

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -33,9 +33,9 @@ jobs:
...
+          - agent: 3.1.7  # renovate: latest juju 2
```
Comment (marceloneppel) [inline .github/workflows/ci.yaml:38]: Shouldn't it be the following?
```suggestion
          - agent: 3.1.7  # renovate: latest juju 3
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (marceloneppel) [reply .github/workflows/ci.yaml:37]: If we pin it, does Renovate update it in its PRs?

Comment (dragomirp) [reply .github/workflows/ci.yaml:37]: Yes, Renovate will update the Juju 2 agent and libjuju 2 in the same PR.

**Review by marceloneppel — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**